contrib/Profiler/Profiler.py

```diff
@@ -0,0 +1,337 @@
+    # return cpuPercent
+    return [containerDelta, sysDelta, onlineCPUs, cpuPercent]
+
+
+def getGpuUtilization(gpu_idx):
+    try:
+        handle = nvmlDeviceGetHandleByIndex(gpu_idx)
+        util = nvmlDeviceGetUtilizationRates(handle)
+        # gpu_util = int(util.gpu)
+    except NVMLError as err:
+        error = handleError(err)
+        # gpu_util = error
+        util = error
+    return util
+
+
+def getGpuMemory(gpu_idx):
+    try:
+        handle = nvmlDeviceGetHandleByIndex(gpu_idx)
+        mem = nvmlDeviceGetMemoryInfo(handle)
+    except NVMLError as err:
+        error = handleError(err)
+        mem = error
+    return mem
+
+
+def getMemoryPercent(filelist):
+    # docker_memory_used_path = "/sys/fs/cgroup/memory/docker/" + str(container_id) + "*/memory.usage_in_bytes"
+    total_memory_path = "/proc/meminfo"
+
+    memory_docker_used = 0.0
+    total_memory = 1.0
+    for filename in filelist:
+        with open(filename, 'r') as f:
+            for line in f:
+                memory_docker_used = int(line)
+
+    with open(total_memory_path, 'r') as f:
+        for line in f:
+            if line.startswith('MemTotal'):
+                lines = line.split()
+                total_memory = int(lines[1]) * 1024
+                break
+    return [memory_docker_used, total_memory]
+
+
+def getDiskReadBytes(filelist):
+    # docker_disk_path = "/sys/fs/cgroup/blkio/docker/" + str(container_id) + "*/blkio.throttle.io_service_bytes"
+    read_bytes = 0
+    write_bytes = 0
+    for filename in filelist:
+        with open(filename, 'r') as f:
+            for line in f:
+                items = line.split()
+                if len(items) != 3 or len(items) != 2:
+                    return -1
+                if items[1] == 'Read':
+                    read_bytes += int(items[1])
+    return read_bytes
+
+
+def getDiskWriteBytes(filelist):
+    # docker_disk_path = "/sys/fs/cgroup/blkio/docker/" + str(container_id) + "*/blkio.throttle.io_service_bytes"
+    write_bytes = 0
+    for filename in filelist:
+        with open(filename, 'r') as f:
+            for line in f:
+                items = line.split()
+                if len(items) != 3 or len(items) != 2:
+                    return -1
+                if items[1] == 'Write':
+                    write_bytes += int(items[1])
+    return write_bytes
+
+
+def getNetworkBytes(filename):
+    receive_bytes, transmit_bytes = 0, 0
+    with open(filename, 'r') as f:
+        for line in f:
+            if len(line.split()) != 17:
+                continue
+            else:
+                items = line.split()
+                receive_bytes += int(items[1])
+                transmit_bytes += int(items[9])
+    return [receive_bytes, transmit_bytes]
+
+
+# The analyze function
+def AnalyzeSamples(sample_list):
+    if len(sample_list) == 0:
+        return
+    count = len(sample_list)
+    min_cpu, min_cpu_idx = sample_list[0]._cpu_usage, 0
```

Comment:
do not access to a protected member directly, change to `sample.cpu_usage` or add a getter method in `Sample` class